app/src/org/commcare/utils/ViewUtils.java

```diff
@@ -8,21 +8,22 @@
 
 public class ViewUtils {
     /**
-     * Displays a SnackBar with the given message and an "OK" button.
-     * The SnackBar will remain visible indefinitely until the "OK" button is pressed.
+     * Displays a SnackBar with the given message and an button with provided text.
+     * The SnackBar will remain visible indefinitely until the button is pressed.
      *
+     * @param btnText         Button text
      * @param view            The view to find a parent from. This view is used to create the SnackBar.
      * @param message         The message text to show in the SnackBar.
      * @param okClickListener The callback to be invoked when the "OK" button is clicked.
      */
-    public static void showSnackBarWithOk(View view, String message, View.OnClickListener okClickListener) {
+    public static void showSnackBarWith(String btnText, View view, String message, View.OnClickListener okClickListener) {
         Snackbar.make(view, message, Snackbar.LENGTH_INDEFINITE)
-                .setAction("OK", okClickListener)
+                .setAction(btnText, okClickListener)
                 .show();
     }
```

Comment:
Can we refactor this function (and its comment) to not reference "ok" for the click listener (i.e. `clickListener` and not `okClickListener`) since the button text is not always "OK".